### PR TITLE
feature: Add support for zxing as barcode scanning lib 

### PR DESCRIPTION
**Pull request by margau**
This is a follow up of #2903 

## Proposed change

For certain QR Codes scanned in low quality, pyzbar has problems detecting them, while this is no problem with zxing. This PR adds the ability to optionally switch to the zxing implementation.

In addition to #2903, zxing-cpp is only installed if the machine type is `x86_64`  for now, as this avoids compatibility problems with the wheel etc.. We can increase this as feasible. The docker image build was successful.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please explain)

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [ ] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Comment by stumpylog** [comment]
Thank you.  I'm actually working with the upstream to hopefully make it possible to support this on more than x64.  It does appear to be a more actively developed project and might be a good complete replacement if the building can be tweaked for our platforms.

In the mean time, while both libraries exist, it would be good to run the barcode testing for both:

```python
try:
    import zxingcpp
    ZXING_AVAILIBLE = True
except ImportError:
    ZXING_AVAILIBLE = False

@pytest.mark.skipif(
    not ZXING_AVAILIBLE,
    reason="No zxingcpp",
)
@override_settings(CONSUMER_BARCODE_SCANNER="ZXING")
class TestBarcodeZxing(TestBarcode):
    pass
```

**Comment by margau** [comment]
I'm currently not sure what codecov does here, according to the CI it seems like the zxing tests are called.

File: src/documents/barcodes.py [hunk 1]
```diff
@@ -83,18 +89,35 @@ def barcode_reader(image: Image) -> List[str]:
...
-                barcodes.append(decoded_barcode)
+            if barcode.text:
+                barcodes.append(barcode.text)
                 logger.debug(
-                    f"Barcode of type {str(barcode.type)} found: {decoded_barcode}",
+                    f"Barcode of type {str(barcode.format)} found: {barcode.text}",
                 )
+    else:
+        logger.warn(
+            f"Invalid Barcode Scanner {scanner_type}, not scanning for barcodes!",
```
Comment (stumpylog) [inline src/documents/barcodes.py:118]: This is good to check, but would be better handled as a system check in `paperless/checks.py`

File: src/documents/barcodes.py [hunk 2]
```diff
@@ -17,7 +17,13 @@
 from pikepdf import Pdf
 from PIL import Image
 from PIL import ImageSequence
-from pyzbar import pyzbar
+
+scanner_type = settings.CONSUMER_BARCODE_SCANNER.upper()
+
+if scanner_type == "PYZBAR":
```
Comment (stumpylog) [inline src/documents/barcodes.py:23]: These imports should move into the matching branches instead

**Review by stumpylog — CHANGES_REQUESTED** [review]

**Comment by stumpylog** [comment]
zxing does locate the image if read directly:
```python
>>> import zxingcpp
>>> from PIL import Image
>>> im = Image.open("src/documents/tests/samples/barcodes/patch-code-t-middle.tiff")
>>> results = zxingcpp.read_barcodes(im)
>>> im.seek(0)
>>> zxingcpp.read_barcodes(im)
[]
>>> im.seek(1)
>>> zxingcpp.read_barcodes(im)
[<zxingcpp.Result object at 0x7f8fa61a2c70>]
```
So there may be something not working correctly in `convert_from_tiff_to_pdf` or around there which can be fixed, instead of skipping tests.

**Comment by margau** [comment]
Hi, is there anything left to do on this MR?

**Comment by stumpylog** [comment]
As there isn't a release pending, I'm going to wait and give the zxing upstream some time to see about incorporating my changes into it, so this can be enabled for all platforms.

**Review by stumpylog — APPROVED** [review]
I'm still hoping to get this for all platforms in the future but the changes here look good at the moment.

**Pull request merged**